fix: refactor model

KaralchukArtem opened the pull request.

Commit:
fix: refactor model

DHrytsuk commented on src/main/java/com/example/finalprojectinterpol/model/CategoryModel.java:
```diff
@@ -7,13 +7,12 @@
-@Table(name = "reportage")
-public class ReportageModel {
+@Table(name = "category")
+public class CategoryModel {
```
Remove `Model` part from all entities

Commit:
feat: added controller, mapper, service, repository for Statement

DHrytsuk commented on src/main/java/com/example/finalprojectinterpol/dto/statement/StatementCreateDTO.java:
```diff
@@ -0,0 +1,32 @@
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
```
Do you need both types of constructor?

DHrytsuk commented on src/main/java/com/example/finalprojectinterpol/service/StatementServiceImpl.java:
```diff
@@ -0,0 +1,42 @@
+@RequiredArgsConstructor
+public class StatementServiceImpl implements StatementService {
+    @Autowired
+    StatementRepository statementRepository;
```
access modifier

Commit:
fiat: create user( rep, entity, ...)

Commit:
create:  Security auth ( login , registration ) user ...
 WebSecurityConfig for User
 MvcConfig (login)
 Role (service,jpa)

Commit:
fix: entity types of constructor, changed access modifier StatementServiceImpl

Commit:
fiat: add Statement ( jsp,controller, security )

DHrytsuk commented on src/main/java/com/example/finalprojectinterpol/service/StatementServiceImpl.java:
```diff
@@ -0,0 +1,42 @@
+@Service
+@RequiredArgsConstructor
+public class StatementServiceImpl implements StatementService {
+    @Autowired
```
You should use either @RequiredArgsConstructor or @Autowired annotation for all fields, not both approaches.
Pay attention, that same approach should be in all classes in project

Commit:
fix: entity types of constructor, changed access modifier StatementServiceImpl

KaralchukArtem committed:
Merge pull request #2 from KaralchukArtem/entity_branch_develop

fiat: create user( rep, entity, ...)

Commit:
feat: StatementServiceImpl ...
fix -> Statement (controller,rep,service )
       User (Service)
       WebSecurityConfig
edit -> refactor String page ModelAndView
add -> jsp (statement) page

Commit:
feat: ...
fix -> Statement (controller,rep,service )
       User (Service,DTO)
       WebSecurityConfig
add -> News (Controller, Service, DTO)

DHrytsuk commented on src/main/java/com/example/finalprojectinterpol/dto/news/NewsCreateDTO.java:
```diff
@@ -0,0 +1,22 @@
+    private String text;
+    private LocalDate year;
+    private User userModel;
+    private Category category_id;
```
It shoul be in camel-case.
And all other fields roo

DHrytsuk commented on src/main/java/com/example/finalprojectinterpol/service/StatementServiceImpl.java:
```diff
@@ -5,18 +5,24 @@
     private StatementRepository statementRepository;
-
+    @Autowired
+    private UserRepository userRepository;
```
We can inject only corresponding repository to the service (eg. StatementServiceImpl -> StatementRepository).
All other repositories should be replaced with corresponding services
